src/mcl.jl

```diff
@@ -69,23 +69,35 @@ function _mcl_clusters(mcl_adj::Matrix{Float64}, allow_singles::Bool, zero_tol::
     clu_ixs, clu_sizes, unassigned_count
 end
 
-# adjacency matrix inflation (element-wise raising to a given power) kernel
-function _mcl_inflate!{T<:Real}(dest::Matrix{Float64}, src::Matrix{T}, inflation::Integer)
-    map!(el -> el^inflation, dest, src)
-end
+# adjacency matrix integral expansion (matrix-wise raising to a given power) kernel
+_mcl_expand{T<:Number}(src::AbstractSparseMatrix{T}, expansion::Integer) = src ^ expansion
```

Comment:
There is no non-integral power function for sparse matrices. So method would and following declaration (below) would prevent usage of non-integral expansion factor and throw `MethodError` in `mcl` rather than somewhere in depths of `Base.LinAlg`.

Reply:
In fact, `speye(5)^2.0` works, so throwing `MethodError` for any non-Integer power might be too aggressive. Also this MethodError would not be overly descriptive. Maybe something like
```julia
function _mcl_expand(src::AbstractMatrix, expansion::Number)
    try
        return src^expansion
    catch ex
        if ex isa MethodError
            throw ArgumentError("MCL expansion of $(typeof(src)) with expansion=$expansion not supported")
        else
            rethrow()
        end
    end
end       
```

Reply:
You're right.

Reply:
I'm also a little bit concerned explicitly enumerating here what matrix types are supported, because the situation in Base may change over time.
For example, `Symmetric` matrices are supported just fine, but with the current two `_mcl_expand` methods they wouldn't be allowed.